test/xpath/test_base.rb

```diff
@@ -416,10 +416,49 @@ def test_preceding
       assert_equal( 4, cs.length )
     end
 
+    def test_preceding_multiple
+      d = REXML::Document.new("<a><b/><c/><d/><d/><e/><f/></a>")
+      matches = REXML::XPath.match(d, "a/d/preceding::node()")
+      assert_equal(["d", "c", "b"], matches.map(&:name))
+    end
+
+    def test_following_multiple
+      d = REXML::Document.new("<a><b/><c/><d/><d/><e/><f/></a>")
+      matches = REXML::XPath.match(d, "a/d/following::node()")
+      assert_equal(["d", "e", "f"], matches.map(&:name))
+    end
+
+    def test_following_sibling
+      d = REXML::Document.new("<a><b><x/><c/><d/></b><b><x/><e/></b></a>")
+      matches = REXML::XPath.match(d, "a/b/x/following-sibling::node()")
+      assert_equal(["c", "d", "e"], matches.map(&:name))
+
+      d = REXML::Document.new("
+      <div>
+        <div>
+          <a/><w/>
+        </div>
+        <a/><x/>
+        <a/><y/>
+        <a/><z/>
+      </div>")
+      # Finds a node flowing <a/>
+      matches = REXML::XPath.match(d, "//a/following-sibling::*[1]")
+      assert_equal(["w", "x", "y", "z"], matches.map(&:name))
```

Comment:
Could you use a separated test with suitable test name?

Reply:
OK.
I see.